src/compiler/checker.ts

```diff
@@ -11449,14 +11463,20 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
     function reportCircularityError(symbol: Symbol) {
         const declaration = symbol.valueDeclaration as VariableLikeDeclaration;

```

Comment:
Given the `if`, maybe this needs to be `VariableLikeDeclaration | undefined`?

Reply:
Or actually, I think this should "just work" without the cast.

Reply:
Should I remove the cast altogether?

Reply:
Yes, I think that should just compile and work nicely.